packages/node_modules/@webex/internal-plugin-conversation/src/conversation.js

```diff
@@ -179,6 +180,111 @@ const Conversation = WebexPlugin.extend({
       });
   },
 
+  /**
+   * generate a deterministic HMAC for a reaction
+   * @param {Object} displayName displayName of reaction we are sending
+   * @param {Object} parent parent activity of reaction
+   * @returns {Promise<HMAC>}
+   */
+  createReactionHmac(displayName, parent) {
+    // not using webex.internal.encryption.getKey() because the JWK it returns does not have a 'k'
+    // property. we need jwk.k to correctly generate the HMAC
+    return this.webex.internal.encryption.unboundedStorage.get(parent.encryptionKeyUrl)
+      .then((keyString) => {
+        const key = JSON.parse(keyString);
+        // when we stringify this object, keys must be in this order to generate same HMAC as
+        // desktop clients
+        const formatjwk = {k: key.jwk.k, kid: key.jwk.kid, kty: key.jwk.kty};
+
+        const source = `${JSON.stringify(formatjwk)}${parent.id}${displayName}`;
+
+        const hmac = crypto
+          .createHmac('sha256', parent.id)
+          .update(source)
+          .digest('hex');
+
+        return Promise.resolve(hmac);
+      });
+  },
+
+  /**
+   * send add or delete reaction to convo service
+   * @param {Object} conversation
+   * @param {Object} reactionPayload
+   * @returns {Promise<Activity>}
+   */
+  sendReaction(conversation, reactionPayload) {
```

Comment:
Can you mark this as `@private` so we don't generate docs for it publicly since this is a helper function?